cli/command/stack/deploy.go

```diff
@@ -26,26 +28,31 @@ func NewDeployCommand(c cli.Interface) *cobra.Command {
 		Use:     "deploy",
 		Aliases: []string{"up", "start"},
 		Short:   "Deploy a stack with a docker compose v3 file",
-		PreRunE: cli.ExactArgs(1),
+		PreRunE: cli.RangeArgs(0, 1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.name = args[0]
-			return deploy(c)
+			return deploy(c, args)
 		},
 	}
 	cmd.Flags().StringVarP(&opts.file, "compose-file", "c", "", "Path to a Compose v3 file")
 	return cmd
 }
 
-func deploy(c cli.Interface) error {
-	c.Console().Printf("Deploying stack %s using %s\n", opts.name, opts.file)
+func deploy(c cli.Interface, args []string) error {
+	var name string
+	if len(args) == 0 {
+		name = strings.Split(strings.TrimSuffix(filepath.Base(opts.file), ".yml"), ".")[0]
```

Comment:
You're assuming the extension will always be `.yml`, but `.yaml` is also valid. I would just take the filename base regardless of extension. If it's not a valid yaml file, the the deploy will fail anyway.